### PR TITLE
feat(TDKN-167):remove refreshProperties and added ref after callback

### PR DESCRIPTION
**Pull request by sgandon**
**What is the current behavior?** (You can also link to an open issue here)
refreshProperties where hastily developed for dataprep cause there was a need to update the current properties state after the Properties is deserialized and references are setup. This is not required.

**What is the new behavior?**
Properties already has a deserialized method to be overriden if necessary but for notification after the references are setup the Studio already use an after callback, so I also implemented it when the references are automatically resolved in daikon. 

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
Only salesforce if using this removed refresh method and I will create a PR to fix this.
The way to go is to create an after<MyReferenceProperties>() method to get notified.

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/84/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/84/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/85/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/85/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/93/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/93/)

File: daikon/src/main/java/org/talend/daikon/properties/PropertiesDynamicMethodHelper.java [hunk 1]
```diff
@@ -117,6 +117,13 @@ static public void afterFormFinish(Properties props, String formName, Repository
         doInvoke(props, findMethod(props, Properties.METHOD_AFTER_FORM_FINISH, formName, REQUIRED), repostory);
     }
 
+    static public void afterReference(Properties props, ReferenceProperties<Properties> refProp) throws Throwable {
```
Comment (gonchar-ivan) [inline daikon/src/main/java/org/talend/daikon/properties/PropertiesDynamicMethodHelper.java]: I used to put "public" modifier before "static". Like this
`public static void`
But I'm not sure whether it is a rule or just a matter of habit.

However, seems there are some recommendation about the question:
http://checkstyle.sourceforge.net/config_modifier.html

Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/properties/PropertiesDynamicMethodHelper.java]: you are righ about it.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/94/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/94/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/95/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/95/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/96/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/96/)

**Pull request merged**